storagedriver/filesystem/driver.go

```diff
@@ -265,6 +265,12 @@ func (d *Driver) Delete(subPath string) error {
 	return err
 }
 
+// URLFor returns a URL which may be used to retrieve the content stored at the given path.
+// May return an UnsupportedMethodErr in certain StorageDriver implementations.
+func (d *Driver) URLFor(path string) (string, error) {
```

Comment:
I am starting to think we need a default implementation of the storage driver, with method stubs and simple implementations for `URLFor`, `PutContent`, `GetContent` and `Move`. Implementors can embed and replace as the driver dictates.